README.md

```diff
@@ -42,67 +42,139 @@ messages if a JSON-document/instance does not comply to its schema.
 
-# How to use
+# Building
 
-The current state of the build-system needs at least version **3.6.0** of NLohmann's
-JSON library. It is looking for the `json.hpp` within a `nlohmann/`-path.
+This library is based on Niels Lohmann's JSON-library and thus has
+a build-dependency to it.
 
-When build the library you need to provide the path to the directory where the include-file
-is located as `nlohmann/json.hpp`.
+Currently at least version **3.6.0** of NLohmann's JSON library
+is required.
 
-## Build
+Various methods using CMake can be used to build this project.
 
-### Within a build-dir
+## Build out-of-source
+
+Do not run cmake inside the source-dir. Rather create a dedicated build-dir:
 
 ```Bash
 git clone https://github.com/pboettch/json-schema-validator.git
 cd json-schema-validator
 mkdir build
 cd build
-cmake .. \
-    -DNLOHMANN_JSON_DIR=<path/to/>nlohmann/json.hpp \
-    -DJSON_SCHEMA_TEST_SUITE_PATH=<path/to/JSON-Schema-test-suite> # optional
-make # install
+cmake [..]
+make
+make install # if needed
 ctest # run unit, non-regression and test-suite tests
 ```
-### As a subdirectory from within
 
-```CMake
-# create an interface-target called json-hpp
-add_library(json-hpp INTERFACE)
-target_include_directories(json-hpp
-    INTERFACE
-        path/to/nlohmann/json.hpp)
+## Building as shared library
 
-# set this path to schema-test-suite to get tests compiled - optional
-set(JSON_SCHEMA_TEST_SUITE_PATH "path/to/json-schema-test-suite")
-enable_testing() # if you want to inherit tests
+By default a static library is built. Shared libraries can be generated by using
+the `BUILD_SHARED_LIBS`-cmake variable:
 
-add_subdirectory(path-to-this-project json-schema-validator)
+In your initial call to cmake simply add:
+```bash
+cmake [..] -DBUILD_SHARED_LIBS=ON [..]
 ```
 
-### Building a shared library
+## Providing access to nlohmann-json
 
-By default a static library is built. Shared libraries are generated by using
-the `BUILD_SHARED_LIBS`-cmake variable:
+The CMake-file of this libraries tries several ways to ultimately include
+`nlohmann/json.hpp`
+
+During the cmake-configurate-step there are 4 tries done trying to
+access nlohmann-json:
+
+1. link with a nlohmann_json::nlohmann_json-target,
+2. link with a nlohmann_json-target,
+3. find the nlohmann_json-cmake-package and link with nlohmann_json::nlohmann_json-target or
+4. find path to `nlohmann/json.hpp`.
+
+1 and 2 are there to make it work then this project is added as
+a sub-directory (via `add_subdirectory()`), 3 and 4 can be
+assisted by setting the `nlohmann_json_DIR`-variable.
+
+### Building as a CMake-subdirectory from within another project
+
+Adding this library as a subdirectory to a parent project is one way of
+building it.
+
+If the parent project
+
+- already used `find_package()` to find the CMake-package of nlohmann_json, method 1 will work.
+- uses the git-repo of nlohmann_json as a subdirectory, method 2 will work.
```

Comment:
method 1 will work when the git-repo of nlohmann_json is used as a subdirectory. Method 2 will never be reached.

Reply:
```suggestion
- uses the git-repo of nlohmann_json as a subdirectory, method 1 will work.
```